### DISCUSSION TITLE
Custom Dataset and DataLoader: list object is not callable

### DISCUSSION THREAD

**Discussion by ahmedramly** [pyg-team/pytorch_geometric#3103] [Q&A] [answered]
Hi, 
I have a dataset that is sorted as (subjects * trials) where each trial is a graph. I wrote a Dataset class to load the data to the DataLoader where the training is on n subjects and the test is on other subjects. the problem is when I load the data in the DataLoader and start training I get this error 
![Screenshot from 2021-09-08 18-22-40](https://user-images.githubusercontent.com/48057708/132538319-ab569488-d7f6-4808-888a-b05c6ae26ec7.png)

the code I used for creating the dataset is this

```
class EEDataset(Dataset):
  def __init__(self, root, folders, transform=None, pre_transform=None):
    self.root = root
    self.folders = folders
    super(EEDataset, self).__init__(root, folders, transform, pre_transform)

  @property
  def raw_file_names(self):
      return []

  @property
  def processed_file_names(self):
    fnames = []
    for folder in self.folders:
      files = os.listdir(os.path.join(self.root, folder))
      nfiles = [os.path.join(folder, f) for f in files]
      fnames.extend(nfiles)
    return fnames

  def len(self):
      return len(self.processed_file_names)

  def get(self, idx):
    subj = torch.load(os.path.join(self.root, self.processed_file_names[idx]))
    return subj
```

and for loading the data

```
from torch_geometric.data import DataLoader

trainset = EEDataset(root_dir,  folders=Dirs[:20])  #Dirs is a list of dir names 'subject ids' 
testset = EEDataset(root_dir,  folders=Dirs[30:40])
train_loader = DataLoader(trainset, batch_size=64, shuffle=True)
test_loader = DataLoader(testset, batch_size=64, shuffle=True)
```
What is possibly wrong with the code that caused this issue?

**Comment by wsad1** [comment, marked as answer]
I think the problem is down to this line `super(EEDataset, self).__init__(root, folders, transform, pre_transform)`. The signature of `__init__` in the `dataset` class is 
```
def __init__(self, root: Optional[str] = None,transform: Optional[Callable] = None,pre_transform:Optional[Callable] = None,pre_filter: Optional[Callable] = None):
```
The second argument is `transform`, in your case the second argument is `folder` that's the reason you get an error when `self.transform` gets executed. Simply removing `folder` from the `super` call should fix the issue.

**Reply by ahmedramly** [reply]
Thanks a lot @wsad1 
It solved my problem
